src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java

```diff
@@ -71,6 +77,20 @@ public SshSlaveLauncher keyCredentials(String username, String key) {
         cred.username.set(username);
         cred.enterDirectly(key);
         cred.add();
+        credentialUsername = username;
         return this;
     }
+
+    /**
+     * Once a credential has been created for a given slave, this method can be used 
+     * to check whether it has already been rendered in the dropdown.
+     */
+    public void waitForCredentialVisible() {
```

Comment:
See discussion [here](https://github.com/jenkinsci/acceptance-test-harness/pull/86#diff-for-comment-57709373) (this same pull request). 

@olivergondza Summing up, it happens that the credentials are added and immediately after the slave config is saved without time for the credential to appear in the combobox, resulting in a slave configuration with `null` credentials. With this, I ensure the credentials are already there and the slave configuration is created with them.